javascript/ql/src/Declarations/UnreachableMethodOverloads.ql

```diff
@@ -0,0 +1,131 @@
+/**
+ * @name Unreachable method overloads
+ * @description Having multiple overloads with the same parameter types in TypeScript
+ *              causes the latter overloads to become unreachable, as the TypeScript compiler
+ *              will always choose the textually first matching overload when selecting
+ *              which overload is called at a callsite.
+ * @kind problem
+ * @problem.severity warning
+ * @id js/unreachable-method-overloads
+ * @precision high
+ * @tags correctness
+ *       typescript
+ */
+
+import javascript
+import semmle.javascript.Variables
```

Comment:
I'm pretty sure you don't need this import.

Reply:
Correct. I've removed it.